logstash-core-event-java/src/main/java/org/logstash/Accessors.java

```diff
@@ -34,10 +34,15 @@ public Object del(String reference) {
                 return ((Map<String, Object>) target).remove(field.getKey());
             } else if (target instanceof List) {
                 int i = Integer.parseInt(field.getKey());
-                if (i < 0 || i >= ((List) target).size()) {
-                    return null;
+                int size = ((List) target).size();
+                if (i >= size || i < -size) {
```

Comment:
seems like there is a bunch of logic on what to do when `i > size`, I think we should add tests for this